### PR TITLE
Restarted Survey register as a node api

### PR DESCRIPTION
**Pull request by SamGodwin2**
This PR wipes the existing functionality from the survey-register and replaces it with a much simpler node api. Play around with it see if anything goes wrong. 

Can use docker-compose up to build the containers.

See https://github.com/ONSdigital/eq-author-app/pull/574 for more info

### CODE REVIEW

**Comment by SamGodwin2** [comment]
[Publish and launch from a registry (9)](https://trello.com/c/8nyY3EJv/1198-publish-and-launch-from-a-registry-9)

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env sh
+
+if [ -n "$SECRETS_S3_BUCKET" ]; then
+    echo "Load Secrets from S3 Bucket [$SECRETS_S3_BUCKET]"
+    aws s3 sync s3://$SECRETS_S3_BUCKET/ /secrets
+fi
+
+yarn start_service
```
Comment (heofs) [inline docker-entrypoint.sh]: newline here

File: index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const express = require("express");
+
+const {
+  getSurveyFromRegistry,
+  getSurveyFromPublisher,
+  insertIntoSurveyRegister
```
Comment (heofs) [inline index.js]: Should this be called `insertIntoSurveyRegistry` to be consistent with `getSurveyFromRegistry`?

File: middleware/index.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const getSurveyFromRegistry = require("./getSurveyFromRegistry");
+const getSurveyFromPublisher = require("./getSurveyFromPublisher");
+const insertIntoSurveyRegister = require("./insertIntoSurveyRegister");
+
+module.exports = {
+  getSurveyFromRegistry,
+  getSurveyFromPublisher,
+  insertIntoSurveyRegister
```
Comment (heofs) [inline middleware/index.js]: Rename this to `insertIntoSurveyRegistry`?

File: README.md [hunk 4]
```diff
@@ -1,23 +1,18 @@
...
-go get -u github.com/golang/dep/cmd/dep
-dep ensure
-go build
-./eq-survey-register
-```
\ No newline at end of file
+
+In order to run this service locally the docker compose file contains all the necessary configuration. Start it using:
+
+`docker-compose up --build`
```
[the remark this answers is not in the document]
Comment (heofs) [reply README.md]: And maybe mention that you need to run yarn first?

File: index.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ const {
 
 const app = express();
 
-app.post(
+app.put(
```
Comment (jonnyshaw89) [inline index.js]: 👍

**Pull request merged**